gprofiler/main.py

```diff
@@ -291,7 +222,7 @@ def _snapshot(self):
             logger.error("Running perf failed; consider running gProfiler with '--perf-mode none' to avoid using perf")
             raise
 
-        if self._runtimes["system"]:
+        if isinstance(self.system_profiler, SystemProfiler):
```

Comment:
Uhh this check is kinda weird. You're trying to check if it's the `NoopProfiler`, right? Maybe add a method `is_noop()` or alternatively, check the opposite condition (`isinstance(.., NoopProfiler)`) which makes more sense when reading.